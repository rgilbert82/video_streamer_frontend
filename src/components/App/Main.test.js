import React from 'react';
import { shallow } from 'enzyme';
import { Main } from '.';

describe(Main, () => {
  let component;

  beforeEach(() => {
    component = shallow(<Main />);
  });

  it('renders properly', () => {
    expect(component).toMatchSnapshot();
  });
});
