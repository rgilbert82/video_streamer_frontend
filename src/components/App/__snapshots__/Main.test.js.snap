// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`class Main extends _react2.default.Component {
  constructor(props) {
    super(props);
  }

  render() {
    return (
      _react2.default.createElement('div', { __source: { fileName: _jsxFileName, lineNumber: 14 }, __self: this },
        _react2.default.createElement(_reactRouterDom.Switch, { __source: { fileName: _jsxFileName, lineNumber: 15 }, __self: this },
          _react2.default.createElement(_reactRouterDom.Route, { exact: true, path: '/', render: () => _react2.default.createElement(_Videos.VideosIndex, { __source: { fileName: _jsxFileName, lineNumber: 16 }, __self: this }), __source: { fileName: _jsxFileName, lineNumber: 16 }, __self: this }),
          _react2.default.createElement(_reactRouterDom.Route, { path: '/videos/:video_id', render: props => _react2.default.createElement(_Videos.VideoPage, Object.assign({}, props, { __source: { fileName: _jsxFileName, lineNumber: 17 }, __self: this })), __source: { fileName: _jsxFileName, lineNumber: 17 }, __self: this }),
          _react2.default.createElement(_reactRouterDom.Route, { exact: true, path: '/users/:user_id', render: () => _react2.default.createElement(_Users.UserPage, { __source: { fileName: _jsxFileName, lineNumber: 18 }, __self: this }), __source: { fileName: _jsxFileName, lineNumber: 18 }, __self: this }),
          _react2.default.createElement(_reactRouterDom.Route, { path: '/*', render: () => _react2.default.createElement(_Misc.NothingHere, { __source: { fileName: _jsxFileName, lineNumber: 19 }, __self: this }), __source: { fileName: _jsxFileName, lineNumber: 19 }, __self: this }))));



  }} renders properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Main />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Switch>
        <Route
                exact={true}
                path="/"
                render={[Function]}
        />
        <Route
                path="/videos/:video_id"
                render={[Function]}
        />
        <Route
                exact={true}
                path="/users/:user_id"
                render={[Function]}
        />
        <Route
                path="/*"
                render={[Function]}
        />
</Switch>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Route
            exact={true}
            path="/"
            render={[Function]}
/>,
          <Route
            path="/videos/:video_id"
            render={[Function]}
/>,
          <Route
            exact={true}
            path="/users/:user_id"
            render={[Function]}
/>,
          <Route
            path="/*"
            render={[Function]}
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "exact": true,
            "path": "/",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "path": "/videos/:video_id",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "exact": true,
            "path": "/users/:user_id",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "path": "/*",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Switch>
          <Route
                    exact={true}
                    path="/"
                    render={[Function]}
          />
          <Route
                    path="/videos/:video_id"
                    render={[Function]}
          />
          <Route
                    exact={true}
                    path="/users/:user_id"
                    render={[Function]}
          />
          <Route
                    path="/*"
                    render={[Function]}
          />
</Switch>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Route
              exact={true}
              path="/"
              render={[Function]}
/>,
            <Route
              path="/videos/:video_id"
              render={[Function]}
/>,
            <Route
              exact={true}
              path="/users/:user_id"
              render={[Function]}
/>,
            <Route
              path="/*"
              render={[Function]}
/>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "exact": true,
              "path": "/",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "path": "/videos/:video_id",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "exact": true,
              "path": "/users/:user_id",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "path": "/*",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
